### DISCUSSION TITLE
Scrum Prep Milestone 1-2

### DISCUSSION THREAD

**Discussion by NatiPerez459** [uprm-inso4101-2024-2025-s1/semester-project-mercado-colegial-online#65] [General]
All team members must comment on the progress they have achieved during this past week or if they have not been able to work. Please do so before our scrum meeting on Friday September 13th.

**Comment by JadStelar** [comment]
Completed an issue on cleaning current repository of unneeded module files and updating .gitignore files, introducing environment variables and tools, such as dotenv, for their usage in the repo for securing api keys and credentials, and documented environment variable standard usage, current .env file contents, and steps to locally download dependencies needed. Documentation can be seen [here](https://docs.google.com/document/d/1Bg0lHtuzFIWX4ny9sntZGf4Mncs9wMP5t9Bogsy2hkQ/edit?usp=sharing). Dependencies for MongoDB have also been added.
-Jahdiel Montero

**Comment by cristian-barreras** [comment]
I have been tasked with the reCAPTCHA, so I have been researching the topic these past few days to implement it in the coming few days.

**Comment by ActuallySeb** [comment]
Installed VueJS Framework for web developers to start designing the website. Aggregated a variety of learning material regarding databases in general and more specific to MongoDB to aid database developers in understanding better the tools they are using.
- Sebastián Montalvo Nieves

**Comment by jonael-garcia26** [comment]
I wasn't able to check GitHub during the first few days of the week, but I did review some information about JavaScript and recently created the issue for email confirmation.

**Comment by JuanDelPueblo** [comment]
Added backend with express.js to receive signup POST requests, contributing to the completion of the now done develop basic sign in task. Assisted teammates in the renaming of an URL route from login to signup and a branch merging task.
~ Edyan Cruz

**Comment by YampierFernandez** [comment]
Created task for establishing the first cluster in Atlas and set up some sample data for future testing. I was later assigned to and completed the same task with another assignee.

**Comment by CovasJJ** [comment]
During this week I was assigned the task of setting up the first Atlas cluster with some sample data for testing with another assignee. The format of the sample data we provided was based on the format seen on the ER diagram. We completed the task during the week.

**Comment by AlvinBadillo** [comment]
This week I was not super hands on as last week I completed various tasks. So this week I have been researching how to implement the controllers, models, and routes for the different entities we have in the database.
- Alvin Badillo

**Comment by PerezMarcelo** [comment]
Researched about **cybersecurity protocols** to make the web page safer, **lazy loading images** for a faster website, and **reCAPTCHA** for a secure experience. Implemented **password guidelines** to improve security, and protection against credential stuffing. Currently working on implementing reCAPTCHA to the sign-up page.

**Comment by Christian2101** [comment]
This week I've created the initial outline for the Milestone #1 Documentation with all the necessary requirements that need to be completed and currently have been  filling out on the requirements the section 2.2.5 Machine requirements.

**Comment by JuanIranzo** [comment]
I created several tasks this week, such as the creation of an interest form for sellers for the next Mercado Colegial iteration, the creation of the Item model for the database, along with it's attributes, the creation of the User model in the database, and the development and implementation of the operation controller for the Item model. I was assigned to the Item model creation task, completed it, and pushed my local branch to the repo (task waiting on review and closure). I also worked with another assignee on the Database research task, in which we researched tutorials for educating the database team. A document was created with a summary of one of the videos, and links to the YouTube videos were also provided, the task has now been closed.

**Comment by franrov4** [comment]
This week, we officially finished setting up our database environment. Due to the work done this week, each team developer has full access/control of the project’s data. 

For upcoming weeks, the plan is to merge with the backend team. We’ve been doing solo work up until now,  connecting the db to the backend of our product, and testing will be our next priorities moving forward.

**Comment by Cabinetman** [comment]
For my contribution this week I completed the mongoDB environment setup by connecting an Atlas account to the project. I am also almost done with completing the user entity model for the database wich I will hopefully have finished by this afternoon.

**Comment by elliot-gomez** [comment]
This week, I focused on researching the most appropriate and recommended format for our required technical documentation. I provided resources and guidelines for the developers responsible for creating the documentation template, as well as instructions to follow. All of this information was compiled into a research report. Additionally, I have been learning how to work with JavaScript and Vue to further contribute to the project.

- Elliot J. Gomez Coriano

**Comment by Kaariinaa08** [comment]
This week, I completed the design for the Seller's Center Page, which includes sections for All Products, Listed, Sold, and Drafts, along with dummy data and a basic statistics section for each tab. I also restructured the tabs under a 'Product Management' section and ensured that the page is prepped for future functionality with navigation buttons to the home and product listing pages.

- Karina Lopez Rodriguez

**Comment by DiegoVazquez1** [comment]
This week i was assigned ot the hashiing and salting of the passwords. i've been doing a lot of research on how and where exactly this code should go and looking at different ways to implement it. It should be done by the end of the weekend

**Comment by gcal77** [comment]
This I week I completed the instructions for the guidelines on the documentation format we will be using (Markdown). I've also been learning to work in Vue in order to start designing the client registration/login page UI.

- Gabriel Calvente Tenorio

**Comment by FabiolaZTorres** [comment]
I was sick this past week, so i was unable to work on any new features. However, to continue contributing to the project's progress, I continued to review team members' work as team leader and merged the `Sign-Up/Login` branch with the `Error Pages` branch.

-Fabiola Z. Torres Maldonado

**Comment by F3rnand0hr** [comment]
This week I was assigned the task of implementing the basic error pages (404 and 500) to our page. I had been doing research this week on how to implement it, and I successfully was able to complete the task. Since we have no home page for the site yet, we have to modify the "Back to Home button" to the correct file for the home page when it is finished.

**Comment by JoshDG03** [comment]
This week I was able to implement 2 methods for the item controller.
